include/IECoreScene/private/PrimitiveAlgoUtils.h

```diff
@@ -35,17 +35,62 @@
 #ifndef IECORESCENE_PRIMITIVEALGOUTILS_H
 #define IECORESCENE_PRIMITIVEALGOUTILS_H
 
+#include "IECoreScene/CurvesPrimitive.h"
+#include "IECoreScene/MeshPrimitive.h"
+#include "IECoreScene/PointsPrimitive.h"
+
+
 #include "IECore/TypeTraits.h"
 
+
 #include "boost/mpl/and.hpp"
 
 #include <numeric>
 
+#include "boost/format.hpp"
+
+#include "tbb/blocked_range.h"
+#include "tbb/parallel_for.h"
+
+#include <unordered_set>
+#include <type_traits>
+
+
 namespace IECoreScene
 {
 namespace Detail
 {
 
+inline int numPrimitives(const IECoreScene::MeshPrimitive *mesh)
+{
+	return mesh->numFaces();
+}
+
+inline int numPrimitives(const IECoreScene::CurvesPrimitive *curves)
+{
+	return curves->numCurves();
+}
+
+inline int numPrimitives(const IECoreScene::PointsPrimitive *points)
+{
+	return points->getNumPoints();
+}
+
+inline IECoreScene::PrimitiveVariable::Interpolation interpolation(const IECoreScene::MeshPrimitive *mesh)
```

Comment:
I don't quite understand what these `interpolation` functions would be used for outside the scope of the segment algorithm. Now that they're in a generic header, perhaps a more specific name would be useful?